core/src/test/java/io/grpc/internal/ManagedChannelImplTest.java

```diff
@@ -4476,6 +4476,38 @@ public String getDefaultScheme() {
     }
   }
 
+  @Test
```

Comment:
This test is not really doing anything to test the code change. The test needs to verify that the service config gets applied when onError gets called.

Reply:
@kannanjgithub nameResolverFactory.allResolved() method internally call the onError() method with managedChannelServiceConfig with our rawServiceConfig details where we are passing the service name as null means it does not have any service name initially so it apply defaultServiceConfig by calling our fix in onError() method.

Reply:
If you just add your test and not add your source code changes, the test still passes. That indicates the test is moot.

Reply:
@kannanjgithub we are further working in this Junit test.

Reply:
@kannanjgithub
we have written the Junit test case and trying to verify the DefaultServiceConfig on call of onError() method but its not working as expected with the below scenario.

Scenario-1:
      nameResolverFactory.nextConfigOrError.set(ConfigOrError.fromConfig(managedChannelServiceConfig));
      nameResolverFactory.resolvers.get(0).listener.onError(Status.UNAVAILABLE);
	 
Scenario-2:
      Status resolutionError = Status.UNAVAILABLE.withDescription("DefaultServiceConfig");
      resolver.listener.onError(null);
	 
Scenario-3:
     nameResolverFactory.resolvers.get(0).listener.onError(Status.UNAVAILABLE);
     FakeNameResolverFactory.FakeNameResolver resolver = nameResolverFactory.resolvers.get(0);
	
Can you please suggest/help how to proceed it further.

Reply:
@kannanjgithub 
Can you please share the update regarding above issue.

Reply:
@kannanjgithub 
Any further updates on the above.